src/main/java/com/google/sps/servlets/ShadowLoadInterviewsServlet.java

```diff
@@ -0,0 +1,210 @@
+        if (!possibleInterview.date().equals(date)) {
+          possibleInterviewsForWeek.add(dayOfSlots);
+          dayOfSlots = new ArrayList<PossibleInterviewSlot>();
+          date = possibleInterview.date();
+        }
+        dayOfSlots.add(possibleInterview);
+      }
+      possibleInterviewsForWeek.add(dayOfSlots);
+    }
+
+    request.setAttribute("weekList", possibleInterviewsForWeek);
+    RequestDispatcher rd = request.getRequestDispatcher("/possibleInterviewTimes.jsp");
+
+    try {
+      rd.forward(request, response);
+    } catch (ServletException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  static List<ScheduledInterview> getPossibleInterviews(
+      ScheduledInterviewDao scheduledInterviewDao,
+      Job position,
+      TimeRange range,
+      PersonDao personDao,
+      String userId) {
+    List<ScheduledInterview> possibleInterviews =
+        scheduledInterviewDao.getForPositionWithoutShadowInRange(
+            position, range.start(), range.end());
+
+    Set<ScheduledInterview> notValidInterviews = new HashSet<ScheduledInterview>();
+    // We want to remove all interviews that the proposed shadow is already involved in or
+    // where either party does not want a shadow.
+    for (ScheduledInterview interview : possibleInterviews) {
+      if (interview.interviewerId().equals(userId)
+          || interview.intervieweeId().equals(userId)
+          || interview.shadowId().equals(userId)
+          || !personDao.get(interview.intervieweeId()).get().okShadow()
+          || !personDao.get(interview.interviewerId()).get().okShadow()) {
+        notValidInterviews.add(interview);
+      }
+    }
+    possibleInterviews.removeAll(notValidInterviews);
+    return possibleInterviews;
+  }
+
+  // Uses an Instant and a timezoneOffset to create a ZonedDateTime instance.
+  private static ZonedDateTime generateDay(Instant instant, ZoneOffset timezoneOffset) {
+    return instant.atZone(ZoneId.ofOffset("UTC", timezoneOffset));
+  }
+
+  // Converts the timezoneOffsetMinutes int into a proper ZoneOffset instance.
+  private static ZoneOffset convertIntToOffset(int timezoneOffsetMinutes) {
+    return ZoneOffset.ofHoursMinutes((timezoneOffsetMinutes / 60), (timezoneOffsetMinutes % 60));
+  }
+
+  private List<PossibleInterviewSlot> scheduledInterviewsToPossibleInterviewSlots(
+      List<ScheduledInterview> interviews, ZoneOffset timezoneOffset) {
+    List<PossibleInterviewSlot> possibleInterviewSlots = new ArrayList<PossibleInterviewSlot>();
+    for (ScheduledInterview interview : interviews) {
+      possibleInterviewSlots.add(
+          PossibleInterviewSlot.create(
+              interview.when().start().toString(),
+              getDate(interview.when().start(), timezoneOffset),
+              getTime(interview.when().start(), timezoneOffset)));
+    }
+    return possibleInterviewSlots;
+  }
+
+  private String getDate(Instant instant, ZoneOffset timezoneOffset) {
```

Comment:
I feel like I've seen this function (and the next two functions) many many times. Is this duplicated from somewhere?

Reply:
I removed the duplication that I saw as obvious, let me know if there is more that I should address.